lib/SIL/Verifier/MemoryLifetimeVerifier.cpp

```diff
@@ -610,9 +610,13 @@ void MemoryLifetimeVerifier::checkBlock(SILBasicBlock *block, Bits &bits) {
       case SILInstructionKind::ValueMetatypeInst:
       case SILInstructionKind::IsUniqueInst:
       case SILInstructionKind::FixLifetimeInst:
-      case SILInstructionKind::DebugValueAddrInst:
         requireBitsSet(bits, I.getOperand(0), &I);
         break;
+      case SILInstructionKind::DebugValueInst:
+        if (cast<DebugValueInst>(&I)->hasAddrVal() &&
+            cast<DebugValueInst>(&I)->exprStartsWithDeref())
+          requireBitsSet(bits, I.getOperand(0), &I);
```

Comment:
Can you explain what's going in here (and perhaps add a comment to that end)?

Reply:
oh yeah, I probably should add comments (after current buildbot finish) about this: Originally this code is
```
case SILInstructionKind::DebugValueAddrInst:
  requireBitsSet(bits, I.getOperand(0), &I);
```
The `requireBitsSet` is simply checking if memory pointed by `I.getOperand(0)` (which is an address-type SSA value) is initialized or not.
When we're migrating to `debug_value` + `op_deref`, the naive refactoring will be:
```
case SILInstructionKind::DebugValueInst:
  if (cast<DebugValueInst>(&I)->hasAddrVal())
    requireBitsSet(...);
```
However, the following SIL snippet will cause `requireBitsSet` to raise a verification error:
```
%a = alloc_stack $Int ...
debug_value %a : $*Int, name ...
```
Because `%a` is not initialized -- but it _doesn't_ need to be initialized, since `debug_value` here is used to specify value declaration (i.e. llvm.dbg.declare-alike) rather than value update.
So the current code is basically saying that we're only doing `requireBitsSet` if the `debug_value` is not used for specifying value declaration.